### ISSUE TITLE
CPPICanTool check inside TOOL:LeftClick might be redundant

### ISSUE THREAD

**Issue by Mista-Tea** [shadowscion/Prop2Mesh#29]
https://github.com/shadowscion/Prop2Mesh/blob/a74d1709499288b7d199cfdadc3869579a305a8e/lua/weapons/gmod_tool/stools/prop2mesh.lua#L368

As a long time implementer of CPPI for my own prop protection, I haven't ran into any other addon that includes a CPPICanTool check inside of TOOL:LeftClick().

TOOL:LeftClick() already causes (or is preceded by) a gamemode CanTool check. Perhaps my setup isn't kosher, but all other addons I've ran on sandbox that cater to CPPI have seemingly never needed CPPICanTool inside LeftClick. I think the assumption is whatever prop protection system that implements CPPI is probably also going to be utilizing the gamemode CanTool hook to allow/prevent TOOL:LeftClick() getting called in the first place.

In my case, this is true. Using P2M on **worldspawn** specifically will be allowed in my CanTool (explicitly returning true), but the redundant CPPICanTool check inside TOOL:LeftClick() will return false because I rely on the initial *CanTool* check to handle whether a player can use a specific tool on things like worldspawn or not. The CPPICanTool check itself doesn't default to allow worldspawn to always return true.

The tl;dr is that players on my server have to now use P2M on a prop that they own in order to spawn the P2M sent, rather than getting to just click on worldspawn. I'd prefer not to change CPPICanTool to always explicitly return true on worldspawn, but that is what the call inside of TOOL:LeftClick() will require me to do.

If my case sounds too specific or deviated from CPPI's intended implementation, I'd understand if you'd prefer to leave the check in. Otherwise I'd argue, from my current perspective, that it is unnecessary and should be handled by CanTool instead.

**Comment by shadowscion** [comment]
That line was intended to prevent players from spawning the entity on props belonging to other players but if it's breaking things like CPPI implementations then I can remove it, not a big deal anyway.

**Comment by shadowscion** [comment]
9dccfe91d24cc8cbe250c226b0694797f2504fc4
